R/PipeOp.R

```diff
@@ -239,6 +239,10 @@ PipeOp = R6Class("PipeOp",
         self$state = NO_OP
         return(named_list(self$output$name, NO_OP))
       }
+      unpacked = unpack_multiplicities(input, multiplicity_type_nesting_level(self$input$train), self$input$name, self$id)
+      if (!is.null(unpacked)) {
+        return(evaluate_multiplicities(self, unpacked, "train", NULL))
```

Comment:
if we always train with an `instate` of `NULL`, then a previously already trained `PipeOp` may be completely reset to a state of `NULL`, if evaluating multiplicities fails during training, see https://github.com/mlr-org/mlr3pipelines/blob/97f38fa14ece48ae887d7bd8e7e1a1d4d7b88386/tests/testthat/test_multiplicities.R#L115

Reply:
What happens with the state of a pipeop when training fails is undefined currently. We could agree on "on failure we reset to `NULL`, because the pipeop is effectively not trained then, but that should probably be implemented and tested separately. Here the `NULL` behaviour is incidental so it probably shouldn't be tested in this PR. you could open an issue (and possibly a PR) about this.